algebra-cl-gen/src/gpu/macros.rs

```diff
@@ -0,0 +1,173 @@
+macro_rules! multiexp_src {
+    (
+        $Limb64:expr, $Curve:ident, false,
```

Comment:
That's exactly what I meant, just one question: Can we parametrize the macros directly with the instance instead of the "import path" ? If we ever refactor the module structure in GingerLib we need to change also here

Reply:
Not sure what you exactly mean. The parameter of macro is just a curve name, without "import path". The entire path added inside the macro. If we will remove it from macro, we will need to add it to the source.rs

Now in source.rs macro looks like this:

        src = multiexp_src!(
            limb64, bn254, true,
            "G1", "Fq", "Fr",
            "G2", "Fq2", "Fr"
        );        

If we will remove paths from macro, same thing will look like this:

        src = multiexp_src!(
            limb64, algebra::curve::bn254, true,
            "G1", "Fq", "Fr",
            "G2", "Fq2", "Fr"
        );        

And actually we will got more duplicated code...

The only thing I have done is moving the import to the head of source.rs file, check this commit please:
https://github.com/HorizenOfficial/ginger-algebra-ocl/commit/c998e561cdf4413d61b6a08c5250e322b3457741